### DISCUSSION TITLE
API Versioned URL outside of api_v1 directory

### DISCUSSION THREAD

**Discussion by mark-meyer** [fastapi/full-stack-fastapi-template#2133] [Questions] [answered]
`/app/api/deps.py` has a hardcoded and versioned url that lives outside the versioning setup of the api.

```
reusable_oauth2 = OAuth2PasswordBearer(
    tokenUrl=f"{settings.API_V1_STR}/login/access-token"
)
```

This is loaded as a dependency of `get_current_user()` which is then imported into the versioned endpoints (i.e. `app/api/api_v1/endpoints/`). Maybe I'm missing something, but this seems like a mistake. It's not clear how one would then add a API_V2 and continue to use this arrangement without duplicating and moving `get_current_user()`. 

Do you have any guidance for this when bumping the api up to the next version?

**Comment by tiangolo** [comment, marked as answer]
You would probably add new endpoints that might not be compatible with the old ones under `/v2/`, but you could still use the same v1 authentication. If you needed a new authentication scheme, you could create that as well.

So, it really depends on your project and your code.
